src/main/java/net/machpi/runelite/influxdb/InfluxDbPlugin.java

```diff
@@ -55,34 +68,62 @@ InfluxDbConfig getConfig(ConfigManager configManager) {
     @Inject
     private MeasurementCreator measurer;
 
+    @Inject
+    private ActivityState activityState;
+
     /**
      * Don't use a shared executor because we don't want to block any game threads.
      */
     private final ScheduledExecutorService executor = new ExecutorServiceExceptionLogger(Executors.newSingleThreadScheduledExecutor());
+    private final EnumMap<Skill, Integer> previousStatXp = new EnumMap<>(Skill.class);
+    private GameState prevGameState;
 
     @Subscribe
     public void onStatChanged(StatChanged statChanged) {
-        if (!config.writeXp()) {
+        int prevStatXp = previousStatXp.getOrDefault(statChanged.getSkill(), 0);
+        if (statChanged.getXp() == prevStatXp)
             return;
+        previousStatXp.put(statChanged.getSkill(), statChanged.getXp());
```

Comment:
This is still triggering xp updates upon login. And sending fake slayer activity.

```suggestion
        Integer prevStatXp = previousStatXp.put(statChanged.getSkill(), statChanged.getXp());
        if (prevStatXp == null || statChanged.getXp() == prevStatXp)
            return;
```

Reply:
That's effectively the same as the current code.  I'm not really sure what's going on here.  I noticed I occasionally got 0-xp values when teleporting/hopping, so maybe I should ignore those events here.

Looking at the discord plugin it discards anything with `previous >= current`, so that would be catching 0-xp values.

Reply:
Your code returns 0 if the value was not found. But when logging in, the xp in the event is total xp for the given skill, not 0.

The suggested code returns a null if the skill hasn't been seen, not 0.

Here's what it looks like when logging in:
```
StatChanged(skill=SLAYER, xp=4276397, level=87, boostedLevel=87)
StatChanged(skill=SLAYER, xp=4276397, level=87, boostedLevel=87)
StatChanged(skill=FARMING, xp=5922814, level=91, boostedLevel=91)
StatChanged(skill=FARMING, xp=5922814, level=91, boostedLevel=91)
StatChanged(skill=RUNECRAFT, xp=2205384, level=81, boostedLevel=81)
StatChanged(skill=RUNECRAFT, xp=2205384, level=81, boostedLevel=81)
StatChanged(skill=HUNTER, xp=2006293, level=80, boostedLevel=80)
StatChanged(skill=HUNTER, xp=2006293, level=80, boostedLevel=80)
...
```

Reply:
And yes, `if (prevStatXp == null || statChanged.getXp() >= prevStatXp)` will catch logged in changes, and 0xp changes.

Reply:
Oh duh -- yeah it would be total XP.  I'll fix it up tomorrow.

Reply:
Okay, made the requested change.  Let me know if it works.

Reply:
10-4. I'll check it out tomorrow